app/api/news/news.get-list.spec.js

```diff
@@ -0,0 +1,209 @@
+      .to.have.status(400)
+      .and.to.have.requestParametersValidationErrors([
+        minimumError({ location: "query", property: "offset", min: "0" }),
+        minimumError({ location: "query", property: "limit", min: "0" }),
+      ])
+      .and.to.matchResponseDocumentation();
+
+    await expectNoSideEffects(app);
+  });
+
+  describe("with default fixtures", () => {
+    let newsOneYearAgo, newsThreeMonthsAgo, newsYesterday;
+    const oneYearAgoDate = new Date(Date.now() - 365 * 24 * 60 * 60 * 1000);
+    const threeMonthsAgoDate = new Date(Date.now() - 90 * 24 * 60 * 60 * 1000);
+    const yesterdayDate = new Date(Date.now() - 24 * 60 * 60 * 1000);
+
+    let initialState;
+    beforeEach(async () => {
+
+
+      // Generate 3 news
+      [newsThreeMonthsAgo, newsYesterday, newsOneYearAgo] = await Promise.all([
+        createNews({
+          created_at: threeMonthsAgoDate,
+          img_url: "http://example.com/2",
+        }),
+        createNews({ created_at: yesterdayDate }),
+        createNews({ created_at: oneYearAgoDate }),
+      ]);
+
+      initialState = await loadInitialState();
+    });
+
+    it("orders news by date", async () => {
+      const req = baseRequest;
+      expect(req).to.matchRequestDocumentation();
+
+      const res = await testHttpRequest(app, req);
+      expect(res)
+        .to.have.status(200)
+        .and.to.have.jsonBody({
+          data: {
+            news: [
+              getExpectedNews(newsYesterday),
+              getExpectedNews(newsThreeMonthsAgo),
+              getExpectedNews(newsOneYearAgo),
+            ],
+          },
+          pagination: {
+            total_records: 3,
+            current_page: 1,
+            page_size: 10,
+            total_pages: 1,
+            links: {
+              prev_page: null,
+              next_page: null,
+            },
+          },
+        })
+        .and.to.matchResponseDocumentation();
+
+      await expectNoSideEffects(app, initialState);
+    });
+
+    it("paginates correctly", async () => {
+      const req = {
+        method: "GET",
+        path: "/news",
+        query: {
+          limit: 1,
+          offset: 1,
+        },
+      };
+      expect(req).to.matchRequestDocumentation();
+
+      const res = await testHttpRequest(app, req);
+
+      expect(res)
+        .to.have.status(200)
+        .and.to.matchResponseDocumentation();
+        
+      expect(res.body.pagination).to.eql({
+          total_records: 3,
+          current_page: 2,
+          page_size: 1,
+          total_pages: 3,
+          links: {
+            prev_page:
+              config.apiUrl +
+              "/news?offset=" +
+              (req.query.offset - req.query.limit) +
+              "&limit=" +
+              req.query.limit,
+            next_page:
+              config.apiUrl +
+              "/news?offset=" +
+              (req.query.offset + req.query.limit) +
+              "&limit=" +
+              req.query.limit,
```

Comment:
maybe hardcode the offset and limit here else it looks like the test is the same code as the controller